### PR TITLE
Fail CI on ESLint warnings

### PR DESCRIPTION
**Pull request by nawatts**
Currently, our custom ESLint rules are configured as warnings instead of errors.

https://github.com/DataBiosphere/terra-ui/blob/6af15805aeed7adc03be7101867d5eb501654b42/.eslintrc.js#L5-L94

Thus, violations of these rules can still be committed because ESLint run in CI only fails on errors. This can add unrelated churn to PRs when other devs notice the warnings and resolve them (for example: https://github.com/DataBiosphere/terra-ui/pull/2862#discussion_r822077233, https://github.com/DataBiosphere/terra-ui/pull/2848#discussion_r823743234).

### CODE REVIEW

**Review by cahrens — APPROVED** [review]
Thanks @nawatts !

File: .circleci/config.yml [hunk 1]
```diff
@@ -195,7 +195,7 @@ jobs:
           paths:
             - node_modules
             - .yarn/cache
-      - run: yarn eslint src
+      - run: yarn eslint --max-warnings=0 src
```
Comment (kyuksel) [inline .circleci/config.yml:198]: @zarsky-broad is this the place we could append `integration-test/src` so test files are linted as well?

```suggestion
      - run: yarn eslint --max-warnings=0 src integration-test/src
```

I also recall seeing another place to update. Was it this [package.json](https://github.com/DataBiosphere/terra-ui/blob/3ada1b41e239c7c954c8ef1e69c68bc218a6ef74/package.json#L62) line?

Comment (nawatts) [reply .circleci/config.yml:198]: If we're adding to the files checked by ESLint, do we want to stop at `integration-tests/src`? Or check all JS files in the project?

Comment (cahrens) [reply .circleci/config.yml:198]: There was concern about things taking a long time if we are linting dependency libraries. I would say lint as much of our own code as you can without introducing speed issues.

Comment (zarsky-broad) [reply .circleci/config.yml:198]: It seemed like we could just put `integration-tests` and it didn't take too long when we tried it

Comment (zarsky-broad) [reply .circleci/config.yml:198]: (or we could be grownups and add an .eslintignore file for node modules and the yarn cache)

**Review by zarsky-broad — APPROVED** [review]
I'm preemptively thumbing this because it's an improvement as is, but please also add it to the script in `package.json`, and feel free to go beyond that as feels appropriate.

File: .circleci/config.yml (hunk 1 above)
Comment (nawatts) [reply .circleci/config.yml:198]: I tried a few runs on my machine. With node_modules and minified JS files ignored, running ESLint on the entire project vs src added ~0.5s to its ~7s second runtime.

Comment (kyuksel) [reply .circleci/config.yml:198]: FWIW I'd be happy to take that penalty 😊

Comment (nawatts) [reply .circleci/config.yml:198]: Great. I'll go ahead and merge this PR (to reduce the chance of additional warnings getting committed to dev while this stays open) and open another one to check everything with ESLint.

Comment (nawatts) [reply .circleci/config.yml:198]: Opened #2873.

**Pull request merged**